NuGet Restore fails if SQL Server Management Studio 21 is installed

### NuGet Product Used

NuGet.exe

### Product Version

6.14.0.116

### Worked before?

_No response_

### Impact

It's more difficult to complete my work

### Repro Steps & Context

1. Install both Visual Studio 2022 and SQL Server Management Studio (SSMS) 21.
2. In a .NET project, run `nuget restore`
3. NuGet will use MSBuild from SSMS instead of Visual Studio, resulting in a missing `Microsoft.CSharp.Core.targets` file or failure to find SDK resolver.

Note that `vswhere` does _not_ include SSMS in its output, so whatever logic it is using may make sense to include in NuGet.exe.

Using the `-MsBuildPath` argument and pointing to Visual Studio (from running ` 'C:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe' -find MSBuild\**\Bin\MSBuild.exe`) does work as expected.

### Verbose Logs

#### Failed to find SDK Resolver

From [EngageSoftware.Dnn.Logging](https://github.com/EngageSoftware/EngageSoftware.Dnn.Logging)

```shell
$ NuGet.exe restore -verbosity detailed
NuGet Version: 6.14.0.116
Restoring NuGet packages for solution D:\code\EngageSoftware.Dnn.Logging\EngageSoftware.Dnn.Logging.sln.
MSBuild auto-detection: using msbuild version '17.14.10.27608' from 'C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
MSBuild P2P timeout [ms]: 120000
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\msbuild.exe "C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\sjxhch3p.fff.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:fa
lse /v:q /p:NuGetPropsFile="C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\2iewiy24.mno.nugetrestore.props" /p:NuGetRestoreTargets="C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\dkassp1n.vdz.nugetrestore.targets" /p:R
estoreUseCustomAfterTargets="True" /p:DisableCheckingDuplicateNuGetItems="True" /p:RestoreTaskAssemblyFile="D:\code\Dnn.Platform\tools\NuGet.CommandLine.6.14.0\tools\NuGet.exe" /p:RestoreSolutionDirectory="D:\code\EngageSoftware.Dnn.Logging\\" /p:SolutionDir="D:\code\EngageSoftware.Dnn.Logging\\" /p:SolutionName="EngageSoftware.Dnn.Logging"
D:\code\EngageSoftware.Dnn.Logging\EngageSoftware.Dnn.Logging\EngageSoftware.Dnn.Logging.csproj : error MSB4247: Could not load SDK Resolver. A manifest file exists, but the path to the SDK Resolver DLL file could not be found. 
Manifest file path 'C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\SdkResolvers\Microsoft.Build.NuGetSdkResolver\Microsoft.Build.NuGetSdkResolver.xml'. SDK resolver path: C:\Program Files\Microsoft SQL Server Management Studio 21\Release\Common7\IDE\CommonExtensions\Microsoft\NuGet\Microsoft.Build.NuGetSdkResolver.dll

NuGet.CommandLine.ExitCodeException: Exception of type 'NuGet.CommandLine.ExitCodeException' was thrown.
   at NuGet.CommandLine.MsBuildUtility.<GetProjectReferencesAsync>d__8.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\MsBuildUtility.cs:line 205
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.RestoreCommand.<GetDependencyGraphSpecAsync>d__71.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\Commands\RestoreCommand.cs:line 800
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.RestoreCommand.<DetermineInputsFromMSBuildAsync>d__66.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\Commands\RestoreCommand.cs:line 666
```

#### Failure to find `Microsoft.CSharp.Core.targets`

From [Dnn.Platform](https://github.com/dnnsoftware/Dnn.Platform)

```shell
$ NuGet.exe restore -verbosity detailed
NuGet Version: 6.14.0.116
Restoring NuGet packages for solution D:\code\Dnn.Platform\DNN_Platform.sln.
MSBuild auto-detection: using msbuild version '17.14.10.27608' from 'C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
MSBuild P2P timeout [ms]: 432000
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\msbuild.exe "C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\qdnadmda.00z.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:fa
lse /v:q /p:NuGetPropsFile="C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\wjf4luyz.ao1.nugetrestore.props" /p:NuGetRestoreTargets="C:\Users\BrianDukes\AppData\Local\Temp\NuGetScratch\ltyu3vrt.enc.nugetrestore.targets" /p:R
estoreUseCustomAfterTargets="True" /p:DisableCheckingDuplicateNuGetItems="True" /p:RestoreTaskAssemblyFile="D:\code\Dnn.Platform\tools\NuGet.CommandLine.6.14.0\tools\NuGet.exe" /p:RestoreSolutionDirectory="D:\code\Dnn.Platform\\" /p:SolutionDir="D:\code\Dnn.Platform\\" /p:SolutionName="DNN_Platform"
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Instrumentation\DotNetNuke.Instrumentation.csproj]  
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Web.Client\DotNetNuke.Web.Client.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Syndication\DotNetNuke.Syndication.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Library\DotNetNuke.Library.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Controls\CountryListBox\CountryListBox.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Web\DotNetNuke.Web.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\HttpModules\DotNetNuke.HttpModules.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\CoreMessaging\DotNetNuke.Modules.CoreMessaging.csproj] 
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\DDRMenu\DotNetNuke.Modules.DDRMenu.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\Groups\DotNetNuke.Modules.Groups.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\HTML\DotNetNuke.Modules.Html.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\Journal\DotNetNuke.Modules.Journal.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\MemberDirectory\DotNetNuke.Modules.MemberDirectory.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\RazorHost\DotNetNuke.Modules.RazorHost.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\AuthenticationProviders\DotNetNuke.Authentication.Facebook\DotNetNuke.Authentication.Facebook.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\AuthenticationProviders\DotNetNuke.Authentication.Google\DotNetNuke.Authentication.Google.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\AuthenticationProviders\DotNetNuke.Authentication.LiveConnect\DotNetNuke.Authentication.LiveConnect.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\AuthenticationProviders\DotNetNuke.Authentication.Twitter\DotNetNuke.Authentication.Twitter.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Utilities\DotNetNuke.Tests.Utilities.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Content\DotNetNuke.Tests.Content.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Data\DotNetNuke.Tests.Data.csproj]      
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.UI\DotNetNuke.Tests.UI.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Core\DotNetNuke.Tests.Core.csproj]      
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Urls\DotNetNuke.Tests.Urls.csproj]      
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.AspNetCCP\DotNetNuke.Tests.AspNetCCP.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Web\DotNetNuke.Tests.Web.csproj]        
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Modules.DDRMenu\DotNetNuke.Tests.Modules.DDRMenu.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Integration\DotNetNuke.Tests.Integration.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Authentication\DotNetNuke.Tests.Authentication.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\HtmlEditorManager\DotNetNuke.Modules.HtmlEditorManager.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Web.Razor\DotNetNuke.Web.Razor.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Web.Mvc\DotNetNuke.Web.Mvc.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Web.Mvc\DotNetNuke.Tests.Web.Mvc.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.Log4net\DotNetNuke.Log4Net.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Dnn.AuthServices.Jwt\Dnn.AuthServices.Jwt.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Admin Modules\Dnn.Modules.Console\Dnn.Modules.Console.csproj]  
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\DnnExportImport\DnnExportImport.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\DnnExportImportLibrary\DnnExportImportLibrary.csproj]  
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\FolderProviders\DotNetNuke.Providers.FolderProviders.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Connectors\Azure\Dnn.AzureConnector.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\ClientCapabilityProviders\Provider.AspNetCCP\DotNetNuke.Providers.AspNetCCP.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DNN.Integration.Test.Framework\DNN.Integration.Test.Framework.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Connectors\GoogleAnalytics\Dnn.GoogleAnalyticsConnector.csproj]
D:\code\Dnn.Platform\DNN Platform\DotNetNuke.ModulePipeline\DotNetNuke.ModulePipeline.csproj : error MSB4247: Could not load SDK Resolver. A manifest file exists, but the path to the SDK Resolver DLL file could not be found. Man
ifest file path 'C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\SdkResolvers\Microsoft.Build.NuGetSdkResolver\Microsoft.Build.NuGetSdkResolver.xml'. SDK resolver path: C:\Program Files\Microsoft SQL Server Management Studio 21\Release\Common7\IDE\CommonExtensions\Microsoft\NuGet\Microsoft.Build.NuGetSdkResolver.dll
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\EditBar\Dnn.EditBar.Library\Dnn.EditBar.Library.csproj] 
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\EditBar\Dnn.EditBar.UI\Dnn.EditBar.UI.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Dnn.PersonaBar.Extensions\Dnn.PersonaBar.Extensions.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Library\Dnn.PersonaBar.Library\Dnn.PersonaBar.Library.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Library\Dnn.PersonaBar.UI\Dnn.PersonaBar.UI.csproj]     
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Tests\Dnn.PersonaBar.Pages.Tests\Dnn.PersonaBar.Pages.Tests.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Tests\Dnn.PersonaBar.Security.Tests\Dnn.PersonaBar.Security.Tests.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Tests\Dnn.PersonaBar.Users.Tests\Dnn.PersonaBar.Users.Tests.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\CachingProviders\DotNetNuke.Providers.Caching.SimpleWebFarmCachingProvider\DotNetNuke.Providers.Caching.SimpleWebFarmCachingProvider.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\ResourceManager\Dnn.Modules.ResourceManager.csproj]    
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\HtmlEditorProviders\DNNConnect.CKE\DNNConnect.CKEditorProvider.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Connectors\GoogleTagManager\Dnn.GoogleTagManagerConnector.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\Dnn.AdminExperience\Tests\Dnn.PersonaBar.ConfigConsole.Tests\Dnn.PersonaBar.ConfigConsole.Tests.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Modules\TelerikRemoval\Dnn.Modules.TelerikRemoval.csproj]      
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Modules\DotNetNuke.Tests.Modules.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Connectors\GoogleAnalytics4\Dnn.GoogleAnalytics4Connector.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\SmtpOAuthProviders\ExchangeOnlineAuthProvider\Dnn.ExchangeOnlineAuthProvider.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Mana
gement Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Providers\SmtpOAuthProviders\GoogleMailAuthProvider\Dnn.GoogleMailAuthProvider.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Tests\DotNetNuke.Tests.Mail\DotNetNuke.Tests.Mail.csproj]      
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.CSharp.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management Studio 
21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets" was not found. Confirm that the expression in the Import declaration "$(CSharpCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.CSharp.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\Website\DotNetNuke.Website.csproj]
C:\Program Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\Microsoft.VisualBasic.CurrentVersion.targets(338,5): error MSB4019: The imported project "C:\Program Files\Microsoft SQL Server Management St
udio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.VisualBasic.Core.targets" was not found. Confirm that the expression in the Import declaration "$(VisualBasicCoreTargetsPath)", which evaluated to "C:\Program Files\Microsoft 
SQL Server Management Studio 21\Release\MSBuild\Current\bin\Roslyn\Microsoft.VisualBasic.Core.targets", is correct, and that the file exists on disk. [D:\code\Dnn.Platform\DNN Platform\DotNetNuke.WebUtility\DotNetNuke.WebUtility.vbproj]
D:\code\Dnn.Platform\DNN Platform\Dnn.ClientSide\Dnn.ClientSide.esproj : error MSB4247: Could not load SDK Resolver. A manifest file exists, but the path to the SDK Resolver DLL file could not be found. Manifest file path 'C:\Pr
ogram Files\Microsoft SQL Server Management Studio 21\Release\MSBuild\Current\bin\SdkResolvers\Microsoft.Build.NuGetSdkResolver\Microsoft.Build.NuGetSdkResolver.xml'. SDK resolver path: C:\Program Files\Microsoft SQL Server Management Studio 21\Release\Common7\IDE\CommonExtensions\Microsoft\NuGet\Microsoft.Build.NuGetSdkResolver.dll

NuGet.CommandLine.ExitCodeException: Exception of type 'NuGet.CommandLine.ExitCodeException' was thrown.
   at NuGet.CommandLine.MsBuildUtility.<GetProjectReferencesAsync>d__8.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\MsBuildUtility.cs:line 205
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.RestoreCommand.<GetDependencyGraphSpecAsync>d__71.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\Commands\RestoreCommand.cs:line 800
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NuGet.CommandLine.RestoreCommand.<DetermineInputsFromMSBuildAsync>d__66.MoveNext() in D:\a\_work\1\s\src\NuGet.Clients\NuGet.CommandLine\Commands\RestoreCommand.cs:line 666
WARNING: Error reading msbuild project information, ensure that your input solution or project file is valid. NETCore and UAP projects will be skipped, only packages.config files will be restored.
All packages listed in packages.config are already installed.
  GET https://api.nuget.org/v3/vulnerabilities/index.json
  OK https://api.nuget.org/v3/vulnerabilities/index.json 128ms
  CACHE https://api.nuget.org/v3/vulnerabilities/index.json
  GET https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/vulnerability.base.json
  GET https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/2025.06.05.05.42.46/vulnerability.update.json
  OK https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/2025.06.05.05.42.46/vulnerability.update.json 26ms
  CACHE https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/2025.06.05.05.42.46/vulnerability.update.json
  OK https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/vulnerability.base.json 73ms
  CACHE https://api.nuget.org/v3-vulnerabilities/2025.06.05.05.42.46/vulnerability.base.json
```

--- a/src/NuGet.Clients/NuGet.CommandLine/MsBuildToolset.cs
+++ b/src/NuGet.Clients/NuGet.CommandLine/MsBuildToolset.cs
@@ -38,16 +38,18 @@ namespace NuGet.CommandLine
             Path = GetMsBuildDirFromVsDir(sxsToolset.GetInstallationPath());
             Version = GetMsBuildVersionFromMsBuildDir(Path);
             InstallDate = ConvertFILETIMEToDateTime(sxsToolset.GetInstallDate());
+            InstallationName = sxsToolset.GetInstallationName();
         }
 
         /// <summary>
         /// This constructor is for testing purposes only
         /// </summary>
-        public MsBuildToolset(string version, string path, DateTime installDate)
+        public MsBuildToolset(string version, string path, DateTime installDate, string installationName)
         {
             Version = version;
             Path = path;
             InstallDate = installDate;
+            InstallationName = installationName;
         }
 
         public Version ParsedVersion
@@ -66,11 +68,13 @@ namespace NuGet.CommandLine
             }
         }
 
-        public bool IsValid => Path != null;
+        public bool IsValid => Path != null && (InstallationName is null || InstallationName.StartsWith("VisualStudio", StringComparison.Ordinal));
 
         public string Version { get; private set; }
 
         public string Path { get; private set; }
+
+        public string InstallationName { get; private set; }
 
         public DateTime InstallDate { get; private set; } = DateTime.MinValue;
 

--- a/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs
+++ b/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs
@@ -54,15 +54,16 @@ namespace NuGet.CommandLine.Test
 
         // Test that GetMsBuildDirectoryInternal deals with invalid toolsets (for example ones created from SKUs that don't ship MSBuild like VS Test Agent SKU) See https://github.com/NuGet/Home/issues/5840 for more info
         [Theory]
-        [MemberData("InvalidToolsetData", MemberType = typeof(ToolsetDataSource))]
-        public void HandlesToolsetsWithInvalidPaths(List<MsBuildToolset> toolsets, string expectedPath)
+        [MemberData(nameof(ToolsetDataSource.InvalidToolsetData), MemberType = typeof(ToolsetDataSource))]
+        public void HandlesToolsetsWithInvalidPaths(List<MsBuildToolset> toolsets, string userVersion, string expectedPath)
         {
             // Arrange
             // Act
+            var orderByDescending = toolsets.OrderByDescending(t => t).ToList();
             var directory = MsBuildUtility.GetMsBuildDirectoryInternal(
-                userVersion: null,
+                userVersion: userVersion,
                 console: null,
-                installedToolsets: toolsets.OrderByDescending(t => t),
+                installedToolsets: orderByDescending,
                 getMsBuildPathInPathVar: (reader) => expectedPath).Path;
 
             // Assert
@@ -414,32 +415,44 @@ namespace NuGet.CommandLine.Test
             private static readonly MsBuildToolset Toolset15_Mon_ShortVersion = new MsBuildToolset(
                 version: "15.1",
                 path: @"c:\vs\25555.00",
-                installDate: new DateTime(2016, 9, 15));
+                installDate: new DateTime(2016, 9, 15),
+                installationName: "VisualStudio/25555.00");
             private static readonly MsBuildToolset Toolset15_Tue_ShortVersion = new MsBuildToolset(
                 version: "15.1",
                 path: @"c:\vs\25555.02",
-                installDate: new DateTime(2016, 9, 16));
+                installDate: new DateTime(2016, 9, 16),
+                installationName: "VisualStudio/25555.02");
             private static readonly MsBuildToolset Toolset15_Wed_ShortVersion = new MsBuildToolset(
                 version: "15.1",
                 path: @"c:\vs\25555.01",
-                installDate: new DateTime(2016, 9, 17));
+                installDate: new DateTime(2016, 9, 17),
+                installationName: "VisualStudio/25555.01");
 
             private static readonly MsBuildToolset Toolset15_Mon_LongVersion = new MsBuildToolset(
                 version: "15.1.137.25382",
                 path: @"c:\vs\25557.00",
-                installDate: new DateTime(2016, 9, 15));
+                installDate: new DateTime(2016, 9, 15),
+                installationName: "VisualStudio/25557.00");
             private static readonly MsBuildToolset Toolset15_Tue_LongVersion = new MsBuildToolset(
                 version: "15.1.137.25382",
                 path: @"c:\vs\25557.02",
-                installDate: new DateTime(2016, 9, 16));
+                installDate: new DateTime(2016, 9, 16),
+                installationName: "VisualStudio/25555.02");
             private static readonly MsBuildToolset Toolset15_Wed_LongVersion = new MsBuildToolset(
                 version: "15.1.137.25382",
                 path: @"c:\vs\25557.01",
-                installDate: new DateTime(2016, 9, 17));
+                installDate: new DateTime(2016, 9, 17),
+                installationName: "VisualStudio/25555.01");
             private static readonly MsBuildToolset InvalidToolsetVSTest = new MsBuildToolset(
                 version: null,
                 path: null,
-                installDate: new DateTime(2017, 9, 7));
+                installDate: new DateTime(2017, 9, 7),
+                installationName: null);
+            private static readonly MsBuildToolset InvalidToolsetVSTest_Ssms = new MsBuildToolset(
+                version: "17.14.14.31908",
+                path: @"c:\ssms\36221.01",
+                installDate: new DateTime(2025, 7, 2),
+                installationName: "SSMS/21.3.7+36221.1");
 
             // Toolset collections
 
@@ -503,10 +516,17 @@ namespace NuGet.CommandLine.Test
                 Toolset4
             };
 
-            private static List<MsBuildToolset> CombinedToolsets_MsBuild15AndVSTestToolsets = new List<MsBuildToolset> {
+            private static List<MsBuildToolset> CombinedToolsets_MsBuild15AndVSTestToolsets =
+            [
                 Toolset15_Wed_LongVersion,
                 InvalidToolsetVSTest
-            };
+            ];
+
+            private static List<MsBuildToolset> CombinedToolsets_MsBuild15AndSsmsToolsets =
+            [
+                Toolset15_Wed_LongVersion,
+                InvalidToolsetVSTest_Ssms
+            ];
 
             // Test data sets
 
@@ -573,10 +593,11 @@ namespace NuGet.CommandLine.Test
                     new object[] { LegacyToolsets_NonNumericVersion, "0" },
                 };
 
-            private static readonly List<object[]> _invalidToolsetData
-                = new List<object[]>
+            private static readonly TheoryData<List<MsBuildToolset>, string, string> _invalidToolsetData
+                = new()
                 {
-                    new object[] { CombinedToolsets_MsBuild15AndVSTestToolsets, Toolset15_Wed_LongVersion.Path}
+                    { CombinedToolsets_MsBuild15AndVSTestToolsets, null, Toolset15_Wed_LongVersion.Path },
+                    { CombinedToolsets_MsBuild15AndSsmsToolsets, "latest", Toolset15_Wed_LongVersion.Path },
                 };
 
             private static readonly List<object[]> _highestPathWithLowVersionMatchData
@@ -595,7 +616,7 @@ namespace NuGet.CommandLine.Test
             public static IEnumerable<object[]> IntegerVersionMatchData => _integerVersionMatchData;
             public static IEnumerable<object[]> NonNumericVersionMatchData => _nonNumericVersionMatchData;
             public static IEnumerable<object[]> NonNumericVersionMatchFailureData => _nonNumericVersionMatchFailureData;
-            public static IEnumerable<object[]> InvalidToolsetData => _invalidToolsetData;
+            public static TheoryData<List<MsBuildToolset>, string, string> InvalidToolsetData => _invalidToolsetData;
             public static IEnumerable<object[]> HighestPathWithLowVersionMatchData => _highestPathWithLowVersionMatchData;
 
         }

